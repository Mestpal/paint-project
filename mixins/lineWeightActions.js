import { mapActions, mapGetters } from 'vuex'

export default {
  computed: {
    ...mapGetters('LineWeightManager', [
      'selectedLineWeight',
      'lineWeightsOptions'
    ])
  },
  methods: {
    ...mapActions('LineWeightManager', [
      'updateLineSelectedWeight'
    ]),
    updateLineWeightAction (lineEvent) {
      this.updateLineSelectedWeight(lineEvent)
    }
  }
}
