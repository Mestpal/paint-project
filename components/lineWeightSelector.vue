<template>
  <v-card>
    <v-card-title> Line Weight </v-card-title>
    <v-card-text>
      <lineOption
        v-for="(lineOption, index) in lineWeightsOptions"
        v-bind:key="index"
        :weight="lineOption"
        @lineEvent="updateLineWeight($event)"
      />
    </v-card-text>
  </v-card>
</template>

<script>
import lineWeightActions from '@/mixins/drawModules/lineWeightActions'
import lineOption from '@/components/atomics/lineOption'

export default {
  components: {
    lineOption
  },
  mixins: [lineWeightActions],
  methods: {
    updateLineWeight (lineEvent) {
      console.log(`Update the line weight to ${lineEvent.weight}`)
      this.updateLineWeightAction(lineEvent)
    }
  }
}
</script>
