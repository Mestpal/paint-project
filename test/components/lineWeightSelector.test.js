import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'

import lineWeightSelector from '@/components/lineWeightSelector'
import { getters } from '@/mixins/drawModules/lineWeightActions'

Vue.use(Vuetify)
Vue.use(Vuex)
const localVue = createLocalVue()

describe('lineWeightSelector.vue', () => {
  let store
  let wrapper

  beforeEach(() => {
    store = new Vuex.Store({
      getters
    })
    wrapper = shallowMount(lineWeightSelector, {
      store,
      localVue
    })
  })

  it('reders the line weight selector component correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  // it('method updateLineWeight', () => {
  //   wrapper = mount(lineWeightSelector, {
  //     store,
  //     methods: {
  //       updateLineWeight () { return jest.fn() }
  //     },
  //     localVue
  //   })
  //   const event = jest.fn()
  //   wrapper.vm.$on('lineEvent', event)

  //   // const lineOptionComponent = wrapper.find(lineOption)
  //   // console.log(lineOptionComponent.selector.methods)
  //   // lineOptionComponent.trigger('lineEvent')

  //   expect(wrapper.vm.updateLineWeight).toHaveBeenCalledTimes(1)
  // })
})
