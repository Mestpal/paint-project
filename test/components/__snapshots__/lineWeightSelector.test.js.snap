// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lineWeightSelector.vue reders the line weight selector component correctly 1`] = `
<v-card-stub loaderheight="4" tag="div">
  <v-card-title-stub> Line Weight </v-card-title-stub>
  <v-card-text-stub></v-card-text-stub>
</v-card-stub>
`;
